backend/molgenis-emx2/src/main/java/org/molgenis/emx2/TableMetadata.java

```diff
@@ -220,7 +220,9 @@ public List<Column> getExpandedColumns(List<Column> columns) {
           if (!ref.isOverlapping()) { // only add overlapping once
             result.put(
                 ref.getName(),
-                new Column(c.getTable(), ref.getName(), true).setType(ref.getPrimitiveType()));
+                new Column(c.getTable(), ref.getName(), true)
+                    .setType(ref.getPrimitiveType())
+                    .setRequired(c.isRequired()));
```

Comment:
and in particular this setRequired was causing the bug

Reply:
suggest to move creation of the expanded ref column to a named function , hard to spot if additional properties need to be copied , could also be a function on the Reference object